linkerd/app/inbound/src/require_identity.rs

```diff
@@ -0,0 +1,73 @@
+use crate::target::TcpAccept;
+use indexmap::IndexSet;
+use linkerd_app_core::{svc::stack::Predicate, Conditional, Error};
+use std::sync::Arc;
+
+/// A connection policy that fails direct connections that don't have a client
+/// identity.
+#[derive(Clone, Debug)]
+pub struct RequireIdentityForDirect;
+
+/// A connection policy that fails connections that don't have a client identity
+/// if they target one of the configured local ports.
+#[derive(Clone, Debug)]
+pub struct RequireIdentityForPorts {
+    ports: Arc<IndexSet<u16>>,
+}
+
+#[derive(Debug)]
+pub struct IdentityRequired(());
```

Comment:
now that we return this in multiple places, should we add a scope to it? or will it be clear from the logs where it came from?

Reply:
it's effectively the same error in both cases. It's as if 4143 is in the required ports list (which i believe it is, anyway).